Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -110,67 +110,71 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
-updateDriverWeeklyZscore ride rideDate weekStartDate weekEndDate driverZscore rideChargeableDistance = do
-  weeklyLeaderBoardConfig <- QLeaderConfig.findLeaderBoardConfigbyType LConfig.WEEKLY >>= fromMaybeM (InternalError "Leaderboard configs not present")
-  (LocalTime _ localTime) <- utcToLocalTime timeZoneIST <$> getCurrentTime
-  let (_, currDayIndex) = sundayStartWeek rideDate
-  let lbExpiry = weeklyLeaderBoardConfig.leaderBoardExpiry - Seconds ((currDayIndex + 1) * 86400) + (Seconds 86400 - secondsFromTimeOfDay localTime) -- Calculated as (total_Seconds_in_week - Week_Day_Index * 86400 + Seconds_Remaining_In_This_Week
-  let currZscore =
-        case driverZscore of
-          Nothing -> weeklyLeaderBoardConfig.zScoreBase + getMeters (fromMaybe 0 rideChargeableDistance)
-          Just zscore -> do
-            let (prevTotalRides, prevTotalDistance) = getRidesAndDistancefromZscore zscore weeklyLeaderBoardConfig.zScoreBase
-            let currTotalRides = prevTotalRides + 1
-            let currTotalDist = prevTotalDistance + fromMaybe 0 rideChargeableDistance
-            currTotalRides * weeklyLeaderBoardConfig.zScoreBase + getMeters currTotalDist
-  Hedis.zAddExp (makeWeeklyDriverLeaderBoardKey weekStartDate weekEndDate) ride.driverId.getId (fromIntegral currZscore) (lbExpiry.getSeconds)
-  let limit = weeklyLeaderBoardConfig.leaderBoardLengthLimit
-  driversListWithScores' <- Hedis.zrevrangeWithscores (makeWeeklyDriverLeaderBoardKey weekStartDate weekEndDate) 0 (limit -1)
-  Hedis.setExp (makeCachedWeeklyDriverLeaderBoardKey weekStartDate weekEndDate) driversListWithScores' (weeklyLeaderBoardConfig.leaderBoardExpiry.getSeconds * weeklyLeaderBoardConfig.numberOfSets)
-
-makeCachedDailyDriverLeaderBoardKey :: Day -> Text
-makeCachedDailyDriverLeaderBoardKey rideDate = "DDLBCK:" <> show rideDate
-
-makeDailyDriverLeaderBoardKey :: Day -> Text
-makeDailyDriverLeaderBoardKey rideDate =
-  "DDLBK:" <> show rideDate
-
-makeCachedWeeklyDriverLeaderBoardKey :: Day -> Day -> Text
-makeCachedWeeklyDriverLeaderBoardKey weekStartDate weekEndDate =
-  "DWLBCK:" <> show weekStartDate <> ":" <> show weekEndDate
-
-makeWeeklyDriverLeaderBoardKey :: Day -> Day -> Text
-makeWeeklyDriverLeaderBoardKey weekStartDate weekEndDate =
-  "DWLBK:" <> show weekStartDate <> ":" <> show weekEndDate
+updateDriverDailyZscore :: (Esq.EsqDBFlow m r, Esq.EsqDBReplicaFlow m r, Metrics.CoreMetrics m, CacheFlow m r, Hedis.HedisFlow m r, MonadFlow m) => Ride.Ride -> Day -> Maybe Double -> Maybe Meters -> Id Merchant -> m ()
+updateDriverDailyZscore ride rideDate driverZscore chargeableDistance merchantId = do
+  dailyLeaderBoardConfig <- QLeaderConfig.findLeaderBoardConfigbyType LConfig.DAILY merchantId >>= fromMaybeM (InternalError "Leaderboard configs not present")
```

Comment:
we can ignore leaderboard for the merchant for which we don't have config in DB instead of throwing error ?